components/tidb_query/src/rpn_expr/impl_json.rs

```diff
@@ -203,6 +203,23 @@ fn json_with_path_validator(expr: &tipb::Expr) -> Result<()> {
     valid_paths(expr)
 }
 
+#[rpn_fn(raw_varg,min_args= 1, max_args = 2, extra_validator = json_with_path_validator)]
+#[inline]
+fn json_keys(args: &[ScalarValueRef]) -> Result<Option<Json>> {
+    assert!(!args.is_empty() && args.len() <= 2);
+    let j: &Option<Json> = args[0].as_ref();
+    let j = match j.as_ref() {
+        None => return Ok(None),
+        Some(j) => j.to_owned(),
+    };
+    let path_expr_list = parse_json_path_list(&args[1..])?;
+    if path_expr_list.is_none() {
+        Ok(None)
+    } else {
+        Ok(j.keys(&path_expr_list.unwrap())?)
+    }
```

Comment:
It's better to use a `if let Some() = ` to avoid calling `unwarp()`
how about 
```
    if let Some(j) = args[0].as_json() {
        if let Some(path_expr_list) = parse_json_path_list(&args[1..])? {
            return Ok(j.keys(&path_expr_list)?);
        }
    }
```

Reply:
Done